src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1685,8 +1685,21 @@ internal bool CheckValueKind(BoundExpression expr, BindValueKind valueKind, Diag
                 case BoundKind.EventAccess:
                     return CheckEventValueKind((BoundEventAccess)expr, valueKind, diagnostics);
                 case BoundKind.DynamicMemberAccess:
+                    {
+                        // dynamic access needs to check if the source of the
+                        // dynamic object is from, for example, a set-only property
+                        var access = (BoundDynamicMemberAccess)expr;
+                        return CheckValueKind(access.Receiver, BindValueKind.RValue, diagnostics);
```

Comment:
Yes. It seems the Value check on the receiver should happen in BindDynamicMemberAccess.
CheckValueKind for BoundDynamicMemberAccess should just return true - since we do not know until runtime.